implementations/rust/ockam/ockam_api/src/nodes/models/portal.rs

```diff
@@ -126,6 +126,29 @@ impl<'a> CreateOutlet<'a> {
     }
 }
 
+/// Request body to delete an outlet
+#[derive(Clone, Debug, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct DeleteOutlet<'a> {
+    /// The alias of the TCP outlet to be deleted
+    #[b(1)] pub alias: CowStr<'a>,
+    #[b(2)] pub addr: CowStr<'a>
```

Comment:
Does this mean the node's TCP outlet does not have to be stopped before removing the alias from the registry?

Also, isn't the alias distinct from the worker address? I figured that was the case from the [create command here](https://github.com/nidnogg/ockam/blob/nidnogg/delete-tcp-outlet/implementations/rust/ockam/ockam_command/src/tcp/outlet/create.rs#L90-L91).

Reply:
> Does this mean the node's TCP outlet does not have to be stopped before removing the alias from the registry?

You still have to stop the outlet using  `node_manager.tcp_transport.stop_outlet()` by passing the outlet worker address, you can retrieve the outlet worker address from the registry using the alias.
The alias is **always** present, if it wasn't specified it would be randomly generated, you check the alias of unnamed outlets using `ockam tcp-outlet list`

> On a related note, isn't the alias distinct from the worker address?

yes, you're right, the alias is distinct from the worker's address,